bool match_node_rdy